document.go

```diff
@@ -18,7 +18,7 @@ type DocumentRef struct {
 }
 
 // NewDocumentRef constructs a DocumentRef with the core two identifiers, ID and Index.
-func NewDocumentRef(id, index string) DocumentRef {
+func NewDocumentRef(index, id string) DocumentRef {
```

Comment:
The `id` and `index` params are out of order WRT the fields in the struct.

Reply:
Unified with index coming first.